playbook/app/pb_kits/playbook/pb_walkthrough/_walkthrough.jsx

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable react/display-name */
+/* eslint-disable no-unused-vars */
+
+/* @flow */
+
+import React, { useState } from 'react'
+import classnames from 'classnames'
+import { buildAriaProps, buildCss, buildDataProps } from '../utilities/props'
+import { globalProps } from '../utilities/globalProps.js'
+import Joyride, { ACTIONS, EVENTS, STATUS } from 'react-joyride'
+import CircleIconButton from '../pb_circle_icon_button/_circle_icon_button'
+import Button from '../pb_button/_button'
+import Card from '../pb_card/_card'
+import Flex from '../pb_flex/_flex'
+import SectionSeparator from '../pb_section_separator/_section_separator'
+import Title from '../pb_title/_title'
+
+type WalkthroughProps = {
+  aria?: object,
+  callback?: () => void,
+  className?: string,
+  continuous?: boolean,
+  data?: object,
+  id?: string,
+  run?: boolean,
+  steps?: array,
+  stepIndex?: number,
+  debug?: Boolean,
+  disableCloseOnEsc?: Boolean,
+  disableOverlay?: Boolean,
+  disableOverlayClose?: Boolean,
+  disableScrolling?: Boolean,
+  floaterProps?: object,
+  hideBackButton?: Boolean,
+  hideCloseButton?: Boolean,
+  showProgress?: Boolean,
+  showSkipButton?: Boolean,
+  spotlightClicks?: Boolean,
+  spotlightPadding?: number,
+  styles?: object,
+}
+
+type TooltipProps = {
+  continuous?: Boolean,
+  className?: String,
+  index?: number,
+  isLastStep?: Boolean,
+  size?: number,
+  step?: object,
+  skip?: Boolean,
+  backProps?: object,
+  closeProps?: object,
+  primaryProps?: object,
+  skipProps?: object,
+  tooltipProps?: object,
```

Comment:
Can you try and describe the shape of these objects here? Would have to do that in Typescript anyway.

Reply:
@thestephenmarshall  can you provide an example of what you mean by shape?

Reply:
So, in Typescript we have `type` as the way we declare the "shape" or expected fields/hierarchy for objects. As an example:

```ts
type closeProps = {
  confirm?: boolean
  onClose: (target: HTMLElement) => void,
}
```

In PropTypes or Flow these are often referred to as "shapes". The `TooltipProps` type above is very basic in terms of what its type interface is telling the developer about the intent of the `Tooltip` component. When it comes to the props: `step`, `backProps` and other similar type fields in your example, simply stating they are `object` doesn't do much at all to explain the interface as a whole.